src/System.Management.Automation/engine/remoting/fanin/PSPrincipal.cs

```diff
@@ -128,12 +125,9 @@ public PSPrincipal UserInfo
 
         /// <summary>
         /// Contains the TimeZone information from the client machine.
+        /// This value is obsolete and is always set to TimeZoneInfo.Local.
         /// </summary>
-        public TimeZoneInfo ClientTimeZone
-        {
-            get;
-            internal set;
-        }
+        public TimeZoneInfo ClientTimeZone => TimeZoneInfo.Local;
```

Comment:
This violates the remoting protocol (PSRP).  We either need to provide the actual client timezone or nothing.

Reply:
How does it violate the protocol? PowerShell, since v3, has ignored whatever was sent to it from the client and did [TimeZone.CurrentTimeZone](https://docs.microsoft.com/en-us/dotnet/api/system.timezone.currenttimezone?view=net-6.0) as per https://github.com/PowerShell/PowerShell/blob/7cc9c874730e87b06dbe54d50e77c1c472996c54/src/System.Management.Automation/engine/remoting/common/WireDataFormat/EncodeAndDecode.cs#L2376-L2392.

The `TimeZone` class is deprecated and further along PowerShell just changes it to [TimeZoneInfo.Local](https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo.local?view=net-6.0)

I can make this `null` and just always return that if you wish, it just seems like having it always set to this value is more aligned with the current behaviour.

Reply:
Yes, please leave it null, as that conforms to the protocol (which makes this capability field optional).  Also I notice that currently this field is null now, with this previous code:

```csharp
#if !CORECLR // TimeZone Not In CoreCLR
                result.TimeZone = TimeZone.CurrentTimeZone;
#endif
```